servicetalk-opentracing-inmemory/src/main/java/io/servicetalk/opentracing/inmemory/KeyValueFormatter.java

```diff
@@ -49,16 +48,17 @@
      * @param verifyExtractedValues {@code true} to make a best effort verification that the extracted values are of the
      * correct format.
      */
-    private TracingHttpHeadersFormatter(boolean verifyExtractedValues) {
+    private KeyValueFormatter(boolean verifyExtractedValues) {
         this.verifyExtractedValues = verifyExtractedValues;
     }
 
-    static InMemorySpanContextFormat<HttpHeaders> traceStateFormatter(boolean validateTraceKeyFormat) {
+    public static InMemorySpanContextFormat<CharSequenceKeyValueAccessor>
+    traceStateFormatter(boolean validateTraceKeyFormat) {
```

Comment:
`CharSequenceKeyValueAccessor` likely requires allocation on each extraction/insertion for pre-existing types (e.g. for servicetalk). An alternative approach is to make an abstract base class with consumer/function to inject/extract values from the carrier type (see below).

```java
abstract class KeyValueFormatter<T> implements InMemorySpanContextFormat<T> {
    private final TriConsumer<T, CharSequence, CharSequence> carrierInjector;
    private final BiFunction<T, CharSequence, CharSequence> carrierExtractor;

    // may make sense to put into concurrent package or somewhere else, not sure
    // if the is an existing interface in the jdk we can leverage...
    interface TriConsumer<X, Y, Z> {
        void accept(X x, Y y, Z z);
    }

   @Override
    public void inject(final InMemorySpanContext context, final T carrier) {
        carrierInjector.accept(carrier, TRACE_ID, context.toTraceId());
        ....
    }

    @Nullable
    @Override
    public InMemorySpanContext extract(final T carrier) {
        CharSequence traceId = carrierExtractor.apply(carrier, TRACE_ID);
        ....
    }
}
```